src/main/drivers/dshot_dpwm.c

```diff
@@ -0,0 +1,196 @@
+#endif
+
+#ifdef USE_DSHOT_DMAR
+FAST_RAM_ZERO_INIT bool useBurstDshot = false;
+#endif
+#ifdef USE_DSHOT_TELEMETRY
+FAST_RAM_ZERO_INIT bool useDshotTelemetry = false;
+#endif
+
+FAST_RAM_ZERO_INIT loadDmaBufferFn *loadDmaBuffer;
+
+FAST_CODE uint8_t loadDmaBufferDshot(uint32_t *dmaBuffer, int stride, uint16_t packet)
+{
+    for (int i = 0; i < 16; i++) {
+        dmaBuffer[i * stride] = (packet & 0x8000) ? MOTOR_BIT_1 : MOTOR_BIT_0;  // MSB first
+        packet <<= 1;
+    }
+    dmaBuffer[16 * stride] = 0;
+    dmaBuffer[17 * stride] = 0;
+
+    return DSHOT_DMA_BUFFER_SIZE;
+}
+
+FAST_CODE uint8_t loadDmaBufferProshot(uint32_t *dmaBuffer, int stride, uint16_t packet)
+{
+    for (int i = 0; i < 4; i++) {
+        dmaBuffer[i * stride] = PROSHOT_BASE_SYMBOL + ((packet & 0xF000) >> 12) * PROSHOT_BIT_WIDTH;  // Most significant nibble first
+        packet <<= 4;   // Shift 4 bits
+    }
+    dmaBuffer[4 * stride] = 0;
+    dmaBuffer[5 * stride] = 0;
+
+    return PROSHOT_DMA_BUFFER_SIZE;
+}
+
+uint32_t getDshotHz(motorPwmProtocolTypes_e pwmProtocolType)
+{
+    switch (pwmProtocolType) {
+    case(PWM_TYPE_PROSHOT1000):
+        return MOTOR_PROSHOT1000_HZ;
+    case(PWM_TYPE_DSHOT1200):
+        return MOTOR_DSHOT1200_HZ;
+    case(PWM_TYPE_DSHOT600):
+        return MOTOR_DSHOT600_HZ;
+    case(PWM_TYPE_DSHOT300):
+        return MOTOR_DSHOT300_HZ;
+    default:
+    case(PWM_TYPE_DSHOT150):
+        return MOTOR_DSHOT150_HZ;
+    }
+}
+
+static void dshotPwmShutdown(void)
+{
+    // DShot signal is only generated if write to motors happen,
+    // and that is prevented by enabled checking at write.
+    // Hence there's no special processing required here.
+    return;
+}
+
+static void dshotPwmDisableMotors(void)
+{
+    // No special processing required
+    return;
+}
+
+static bool dshotPwmEnableMotors(void)
+{
+    // No special processing required
+    return true;
+}
+
+extern void pwmWriteDshotInt(uint8_t index, uint16_t value); // XXX Do this right
+
+static FAST_CODE void dshotWriteInt(uint8_t index, uint16_t value)
+{
+    pwmWriteDshotInt(index, value);
+}
+
+static FAST_CODE void dshotWrite(uint8_t index, float value)
+{
+    pwmWriteInt(index, lrintf(value));
+}
+
+static motorVTable_t dshotPwmVTable = {
+    .enable = dshotPwmEnableMotors,
+    .disable = dshotPwmDisableMotors,
+    .updateStart = motorUpdateStartNull, // May be updated after copying
+    .write = dshotWrite,
+    .writeInt = dshotWriteInt,
+    .updateComplete = pwmCompleteDshotMotorUpdate,
+    .convertExternalToMotor = dshotConvertFromExternal,
+    .convertMotorToExternal = dshotConvertToExternal,
+    .shutdown = dshotPwmShutdown,
+};
+
+FAST_RAM_ZERO_INIT motorDevice_t dshotPwmDevice;
+
+motorDevice_t *dshotPwmDevInit(const motorDevConfig_t *motorConfig, uint16_t idlePulse, uint8_t motorCount, bool useUnsyncedPwm) {
+    UNUSED(idlePulse);
+    UNUSED(useUnsyncedPwm);
+
+    memcpy(&dshotPwmDevice.vTable, &dshotPwmVTable, sizeof(motorVTable_t));
```

Comment:
better use `sizeof(dshotPwmDevice.vTable)`. Also you can simply use `=` operator ...

Reply:
I'll use '='.